src/App.tsx

```diff
@@ -15,36 +16,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_FIELD_ALPH = 'alphabetical';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(
+  goods: string[],
+  { sortType, isReverse }: { sortType: string; isReverse: boolean },
+) {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SORT_FIELD_ALPH:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
```

Comment:
Direct interaction with the DOM is not recommended in React for sorting and reversing the goods list. Instead, it would be better to use the React state and effects to handle sorting and reversing to maintain a declarative approach.